packages/block-editor/src/store/selectors.js

```diff
@@ -1663,21 +1668,19 @@ export function canInsertBlocks( state, clientIds, rootClientId = null ) {
  */
 export function canRemoveBlock( state, clientId, rootClientId = null ) {
 	const attributes = getBlockAttributes( state, clientId );
-
-	// attributes can be null if the block is already deleted.
 	if ( attributes === null ) {
 		return true;
 	}
-
-	const { lock } = attributes;
-	const parentIsLocked = !! getTemplateLock( state, rootClientId );
-	// If we don't have a lock on the blockType level, we defer to the parent templateLock.
-	if ( lock === undefined || lock?.remove === undefined ) {
-		return ! parentIsLocked;
+	if ( attributes.lock?.remove ) {
+		return false;
 	}
-
-	// When remove is true, it means we cannot remove it.
-	return ! lock?.remove;
+	if ( getTemplateLock( state, rootClientId ) ) {
+		return false;
+	}
+	if ( getBlockEditingMode( state, rootClientId ) === 'disabled' ) {
+		return false;
+	}
+	return true;
```

Comment:
What you have is easier to read, but this also works:

```js
	return (
		! attributes ||
		( ! attributes.lock?.remove &&
			! getTemplateLock( state, rootClientId ) &&
			getBlockEditingMode( state, rootClientId ) !== 'disabled' )
	);
```

I don't know why I even wrote that. To look clever I suppose.

Reply:
You look clever!